scenario_gym/metrics/trajectory.py

```diff
@@ -64,3 +67,813 @@ def _step(self, state: State) -> None:
+        return [False, None]
+
+    def _step(self, state: State) -> None:
+        """Predict the delta_v for each entity potentially involved in a collision with ego vehicle."""
+        # Get entities within 10m of the ego, if entities exist... 
+        if len(state.get_entities_in_radius(*state.poses[self.ego][:2], 10)) > 1:
+            # Then loop through non_ego entities and check if they will reach intersection point at approx. the same time as the ego (function _will_collide does this).
+            for entity in state.get_entities_in_radius(*state.poses[self.ego][:2], 10):
+                if entity == self.ego:
+                    continue
+
+                # Get ego vehicle pose, velocity, and bounding box. 
+                pos_ego = state.poses[self.ego][:2]
+                vel_ego = state.velocities[self.ego][:2]
+                box_ego = self.ego.get_bounding_box_geom(state.poses[self.ego])
+                # Get non_ego vehicle pose, velocity, and bounding box.
+                pos_non_ego = state.poses[entity][:2]
+                vel_non_ego = state.velocities[entity][:2]
+                box_non_ego = entity.get_bounding_box_geom(state.poses[entity])
+
+                if entity.__class__.__name__ == 'Pedestrian':
+                    distance_tolerance = 1
+                elif entity.__class__.__name__ == 'Vehicle':
+                    distance_tolerance = 0.1
+
+                # Check if ego will collide with other entity. 
+                self.ego_time_to_collision_bool_and_value = self._will_collide_check(pos_ego, vel_ego, box_ego, pos_non_ego, vel_non_ego, box_non_ego, distance_tolerance)
+
+                # If they will collide, then retrieve time-to-collision. 
+                if self.ego_time_to_collision_bool_and_value[0] == False:
+                    continue
+
+                elif entity.__class__.__name__ != 'Pedestrian' and entity.__class__.__name__ != 'Vehicle':
+                    print(entity.__class__.__name__)
+                    continue
+
+                elif self.ego_time_to_collision_bool_and_value[1] < 1:
+                    ego_class = self.ego.__class__.__name__
+                    entity_class = entity.__class__.__name__
+                    entity_delta_v = (self.predefined_masses[ego_class] / (self.predefined_masses[ego_class] + self.predefined_masses[entity_class])) * (np.linalg.norm(vel_ego) - np.linalg.norm(vel_non_ego))
+
+                    # state.t inclusion in the dictionary (line below). To track/timestamp TTC in simulation.
+                    if state.t not in self.delta_v_dict:
+                        self.delta_v_dict[state.t] = []
+                    self.delta_v_dict[state.t].append({"delta_v" : {f"{entity}": entity_delta_v}})
+                else:
+                    continue 
+        
+    def get_state(self):
+        """Return delta_v dictionary."""
+        return self.delta_v_dict
+    
+
+# #TODOASAP - CONFLICT INDEX
+# class ConflictIndex(Metric):
```

Comment:
Could you please either remove this commented code or turn it into a proper class with a docstring describing what it will do and a `raise NotImplemented`?